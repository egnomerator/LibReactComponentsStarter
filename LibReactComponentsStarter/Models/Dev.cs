namespace LibReactComponentsStarter.Models
{
    public class Dev
    {
        public long Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
    }
}
