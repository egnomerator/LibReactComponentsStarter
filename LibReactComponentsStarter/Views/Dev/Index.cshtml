@{
    ViewData["Title"] = "Devs";
}

<div class="row">
    <div class="col-md-12">
        <div id="reactTableContainer"></div>
    </div>
</div>
@section scripts{
    <script src="~/js/Devs/devsWebAPI.js" asp-append-version="true"></script>
    <script src="~/js/Devs/index.js" asp-append-version="true"></script>
    <script>
        reactDevs.renderReactDevs();
    </script>
}
