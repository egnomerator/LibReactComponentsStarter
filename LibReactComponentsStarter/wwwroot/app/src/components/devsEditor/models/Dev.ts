export interface Dev {
    id: number;
    firstName: string;
    lastName: string;
}
