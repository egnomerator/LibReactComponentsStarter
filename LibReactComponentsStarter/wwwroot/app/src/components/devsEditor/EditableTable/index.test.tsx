import * as React from "react";
import * as ReactTestRenderer from "react-test-renderer";
import { EditableTable } from ".";
import { Dev } from "../models/Dev";

describe("EditableTable", () => {
    it("should render with zero devs", () => {
        const mockView = () => { };
        const mockEdit = () => { };
        const mockDelete = () => { };
        const devs = [];

        const editableTable = ReactTestRenderer.create(<EditableTable devs={devs} view={mockView} edit={mockEdit} delete={mockDelete} />);

        expect(editableTable).toMatchSnapshot();
    });
    it("should render with one dev", () => {
        const mockView = () => { };
        const mockEdit = () => { };
        const mockDelete = () => { };
        const devs: Dev[] = [{ id: 1, firstName: "first", lastName: "last" }];

        const editableTable = ReactTestRenderer.create(<EditableTable devs={devs} view={mockView} edit={mockEdit} delete={mockDelete} />);

        expect(editableTable).toMatchSnapshot();
    });
    it("should render with two devs", () => {
        const mockView = () => { };
        const mockEdit = () => { };
        const mockDelete = () => { };
        const devs: Dev[] = [
            { id: 1, firstName: "first", lastName: "last" },
            { id: 2, firstName: "first", lastName: "last" }
        ];

        const editableTable = ReactTestRenderer.create(<EditableTable devs={devs} view={mockView} edit={mockEdit} delete={mockDelete} />);

        expect(editableTable).toMatchSnapshot();
    });
    it("should render with three devs", () => {
        const mockView = () => { };
        const mockEdit = () => { };
        const mockDelete = () => { };
        const devs: Dev[] = [
            { id: 1, firstName: "first", lastName: "last" },
            { id: 2, firstName: "first", lastName: "last" },
            { id: 3, firstName: "first", lastName: "last" }
        ];

        const editableTable = ReactTestRenderer.create(<EditableTable devs={devs} view={mockView} edit={mockEdit} delete={mockDelete} />);

        expect(editableTable).toMatchSnapshot();
    });
});