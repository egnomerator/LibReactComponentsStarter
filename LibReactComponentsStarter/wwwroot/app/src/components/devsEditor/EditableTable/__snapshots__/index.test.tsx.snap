// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTable should render with one dev 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <td>
        Id
      </td>
      <td>
        Name
      </td>
      <td>
        Action
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          Details
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EditableTable should render with three devs 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <td>
        Id
      </td>
      <td>
        Name
      </td>
      <td>
        Action
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          Details
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          Details
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          Details
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EditableTable should render with two devs 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <td>
        Id
      </td>
      <td>
        Name
      </td>
      <td>
        Action
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          Details
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        <input
          onChange={[Function]}
          type="text"
          value="first"
        />
      </td>
      <td>
        <button
          onClick={[Function]}
        >
          Details
        </button>
         
        <button
          onClick={[Function]}
        >
          Edit
        </button>
         
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EditableTable should render with zero devs 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <td>
        Id
      </td>
      <td>
        Name
      </td>
      <td>
        Action
      </td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="text-center"
        colSpan={3}
      >
        There are no devs.
      </td>
    </tr>
  </tbody>
</table>
`;
